File: API/Models/Estoque.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ABrechozeiraApp.Models;

public partial class Estoque
{
    public int Id { get; set; }

    public int? CodigoEstoque { get; set; }

    [ForeignKey("Produto")]
    public int ProdutoId { get; set; }
    public Produto? Produto { get; set; }
    public int Quantidade { get; set; }
    [StringLength(100)]
    public string? Localizacao { get; set; }

    public DateTime? DataAlteracao { get; set; }

    [ForeignKey("Usuario")]
    public int UsuarioModificacaoId { get; set; }
    public Usuario? UsuarioModificacao { get; set; }

}

File: FRONT/src/app/modules/estoque/models/estoque.ts
export interface Estoque {
    id: number;
    codigoEstoque: string;
    quantidade: number;
    localizacao: string;
    dataAlteracao: Date;
    produtoId: number;
    descricao: string;
    usuarioModificacaoId: number;
    nome: string;
}

export interface EstoqueCreate {
    codigoEstoque: string;
    quantidade: number;
    localizacao: string;
    produtoId: number;
    usuarioModificacaoId: number;
}

export interface EstoqueDTO extends Estoque {
    produto?: {
        descricao: string;
    };
    usuarioModificacao?: {
        nome: string;
    };
} 